_posts/2023-03-09-solidity-developer-survey-2022-results.md

```diff
@@ -0,0 +1,451 @@
+
+Initially, 4.7% said yes. However, when asked which one they were affected by, only two out of the 63 people were able to point to actual Solidity vulnerabilities in the follow-up question. This gives room to assume the actual number of affected developers from this survey is significantly lower than 4.7%, and some respondents may have simply misunderstood the question. We will try to phrase this question more precisely in the next survey.
+
+![High Impact Bugs](/img/2023/02/security_bugs.png)
+
+## Language Design & Upcoming Features
+
+### Favorite Feature / Solidity Aspect
+
+Respondents most like Solidity's syntax, the simplicity with regards to learning, reading, coding and compiling and static typing.
+
+The most mentioned liked features in descending order were:
+
+* Syntax
+* Easy to... read/code/compile/learn
+* Simplicity
+* Static/strong/strict typing
+* Similarity to other languages (most mentioned JS/TS and object-orientedness, also mentioned: Rust, C++, Python)
+* Modifiers
+* Inline assembly
+* Mappings
+* Inheritance
+* (User-defined) types
+* SafeMath / checked and unchecked
+* Yul
+
+### Most Dreaded Aspect
+
+This year, we asked the survey audience a slightly different question: “If you could change one thing about Solidity, what would it be?”
+
+Most mentioned “change requests” in descending order were:
+
+* Fix stack-too-deep
+* Better array handling
+* Gas optimizations
+* Add decimal math (fixed point types / floating types)
+* Better error handling, descriptions
+* Better debugging
+* Higher contact bytecode size limit
+
+### Future Features
+
+#### Most Anticipated Feature
+
+Support for decimal numbers and generics was mentioned most often as the “most anticipated Solidity feature”.
+
+_⚠️ Similar to the previous year, we noted that respondents were using various different terms like "floats", "floating point arithmetic", "floating point number", "fixed point numbers", and "fixed point math". We categorized those as "decimal numbers"._
+
+Most mentioned anticipated features in descending order:
+
+* Support for decimal numbers (fixed point types / floating points)
+* Generics
+* Better optimization
+* No stack-too-deep
+* Better debugging
+* Transient storage
+* Standard library
+* Better error messages
+
+#### EIP Support
+
+We also wanted to know what Solidity-related EIPs the survey respondents think need support in the compiler.
+
+It’s interesting to note that only 93 out of 1401 respondents gave meaningful answers to this question.
+
+EIP-2335 “‘Diamonds, Multi-Facet Proxy” was mentioned most often, followed by EIP-1153 “Transient Storage” and EIP-3540 “EOF - EVM Object Format”.
+
+![Solidity-related EIPs](/img/2023/02/eip_support.png)
```

Comment:
Broken link.

Reply:
If you go to github preview, it looks like this:

![solidity-related-eips-broken-link](https://user-images.githubusercontent.com/137030/224037872-c7b56953-429d-4547-9cb4-02fdc9a7c385.png)

And links to https://github.com/ethereum/solidity-blog/blob/solidity-developer-survey-2022-results/img/2023/02/eip_support.png, which gives me HTTP 404.

Probably it should be `img/2023/...` instead of `/img/2023/...` (i.e. a relative link). I think it works for you on a local server because it serves the post at the root. On github you have the branch name at the beginning of the URL and an absolute link ignores it.

Reply:
should be fixed now! the file had "EIP" in capital and the link only used "eip" in small letters.